multimap/src/main/java/org/infinispan/multimap/impl/ListBucket.java

```diff
@@ -360,18 +349,19 @@ public List<ScoredValue<V>> sort(SortOptions sortOptions) {
       return sort(scoredValueStream, sortOptions);
    }
 
-   public void replace(List<V> list) {
-      this.values.clear();
+   public ListBucket<V> replace(List<V> list) {
       if (list != null && !list.isEmpty()) {
-         this.values.addAll(list);
+         return new ListBucket<>(new ArrayDeque<>(list));
```

Comment:
Maybe check if impl is a Deque and pass directly if so? This makes me wonder, should we trying to pass Deques directly anyways?

Reply:
Changing to only accept Dequeue should be doable. It is only called by one function.